ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -296,6 +304,20 @@ func getResourceFunc(ctx context.Context, key client.ObjectKey, obj client.Objec
 	}
 }
 
+func getNumWorkersToDeleteFunc(ctx context.Context, raycluster *v1alpha1.RayCluster) func() (int, error) {
+	return func() (int, error) {
+		key := client.ObjectKey{Name: raycluster.Name, Namespace: "default"}
+		if err := k8sClient.Get(ctx, key, raycluster); err != nil {
```

Comment:
Just curious, is it a fake Kubernetes client?

Reply:
Yep.

Reply:
`k8sClient` seems to be initialized in **suite_test.go#L77**. It looks like it uses `sigs.k8s.io/controller-runtime/pkg/client` to initialize it. I cannot find the implementation of the fake client. I am not familiar with the test framework in Go. Do you mind pointing it out? Thank you!

https://github.com/ray-project/kuberay/blob/60de974b232eabcf8b9bcbd21d8f3c197ae52b20/ray-operator/controllers/ray/suite_test.go#L77

Reply:
I actually have no idea either -- I just copied exist logic in this test file.
We can try to figure it out tomorrow, though.